components/site/src/lib.rs

```diff
@@ -784,6 +764,41 @@ impl Site {
         Ok(())
     }
 
+    pub fn render_all_feeds(&self) -> Result<()> {
+        let mut start = Instant::now();
+        let library = self.library.read().unwrap();
+
+        for (code, language) in &self.config.languages {
+            let is_default_language = code == &self.config.default_language;
+
+            let skip_default_language_feed_generation =
+                is_default_language && !self.config.generate_feeds && !language.generate_feeds;
```

Comment:
should that ambiguity be resolved when loading the config? Eg error if the value at the top level and in a default language level differs.
This way you can now just do something like `self.config.languages.iter().filter(|x| x.generate_feeds)` for the loop